### PR TITLE
Separate install / access of fie to different files to avoid circular dependencies

### PR DESCRIPTION
**Pull request by lannka**
This is a follow up of https://github.com/ampproject/amphtml/pull/23493#issuecomment-515138398

- moved `isInFie()` to `dom.js`
- moved `getFriendlyIframeEmbedOptional ()` to `services.js`
- removed the dups of the 2 functions.

### CODE REVIEW

**Comment by lannka** [comment]
@choumx let me know if you think there could be a better place for `getFriendlyIframeEmbedOptional(iframe)`

File: src/dom.js [hunk 1]
```diff
@@ -240,6 +240,17 @@ export function isShadowRoot(value) {
...
+/**
+ * @param {!Element} element
+ * @return {boolean}
+ */
+export function isInFie(element) {
+  return (
+    element.classList.contains('i-amphtml-fie') ||
+    !!closestAncestorElementBySelector(element, '.i-amphtml-fie')
+  );
+}
```
Comment (choumx) [inline src/dom.js]: I'd move this to iframe-helper.js too. Keeps FIE functions together.

File: src/iframe-helper.js [hunk 2]
```diff
@@ -586,3 +586,20 @@ export function canInspectWindow(win) {
     return false;
   }
 }
+
+/** @const {string} */
+export const EMBED_PROP = '__AMP_EMBED__';
```
Comment (choumx) [inline src/iframe-helper.js]: Minor: Rename to `FIE_EMBED_PROP` now that this is visible.

File: src/dom.js (hunk 1 above)
Comment (lannka) [reply src/dom.js]: SG done

File: src/iframe-helper.js (hunk 2 above)
Comment (lannka) [reply src/iframe-helper.js]: done

**Pull request merged**